tests/db/migration_tests/dump_tool.py

```diff
@@ -200,6 +201,15 @@ async def test_dump_db(server, client, postgres_db, database_name):
     )
     assert result.code == 200
 
+    # create an orphan resource
+    version = 2
+    path = "/tmp/test" + str(2)
+    key = "std::File[localhost,path=" + path + "]"
+    res2_v1 = data.Resource.new(
```

Comment:
this setup should always use the API, never the DB directly. In the next version, the single resource is not valid (it always would have a corresponding presistent state) . To ensure consistent data, the actual code should make it.

I think making the model respond to an env var and calling `request_recompile` on the `CompilerService` directly is the easiest way.

Reply:
I will need more information about this as I don't understand what I'm doing.
I tried to add this at the start of the test:
```
    await compilerslice.request_recompile(
        env=env1, force_update=False, do_export=False, remote_id=remote_id1, env_vars={"my_unique_var": "1"}
    )
```
and 
to change to model:
```
agent = std::AgentConfig(agentname=host.name, autostart=true)
host = std::Host(name="localhost", os=std::linux, requires=agent)
file = std::ConfigFile(host=host, path="/tmp/test", content="1234", requires=agent)
val = std::get_env(name="my_unique_var",default_value="0")
if val == "1":
    file2 = std::ConfigFile(host=host, path="/tmp/test_orphan", content="test", requires=agent)
end

```
I was expecting to see an additional File resource when the the env var is set to 1 but I see no new resource in the DB.

Reply:
Your compile request has `do_export=False` set. This means don't export the resources to the server.